---
title: "Using mitch to analyse Seurat processed scRNA-seq data"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output: html_document
---

Source: https://github.com/markziemann/mitch_example_workflows

# Introduction

In this workflow I'm following the steps described in the Seurat tutorial called ifnb (April 2020 version):
https://satijalab.org/seurat/v3.1/immune_alignment.html

Then I apply mitch to the output of Seurat to perform enrichment analysis.

Running R4, mitch v1.0.4 and Seurat_3.1.5

```{r}
#devtools::install_github('satijalab/seurat-data')
#devtools::install_github('satijalab/seurat-wrappers')
#devtools::install_github('satijalab/ifnb.SeuratData_3.0.0.tar.gz')
#download.file("https://seurat.nygenome.org/src/contrib/ifnb.SeuratData_3.0.0.tar.gz",destfile="ifnb.SeuratData_3.0.0.tar.gz")
#install.packages("ifnb.SeuratData_3.0.0.tar.gz", repos = NULL, type = "source")
library(ggplot2)
library(Seurat)
library(SeuratData)
library(SeuratWrappers)
library(cowplot)
library(patchwork)
library("ifnb.SeuratData")
data("ifnb")
library("mitch")

ifnb.list <- SplitObject(ifnb, split.by = "stim")

ifnb.list <- lapply(X = ifnb.list, FUN = function(x) {
    x <- NormalizeData(x)
    x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})
immune.anchors <- FindIntegrationAnchors(object.list = ifnb.list, dims = 1:20)
immune.combined <- IntegrateData(anchorset = immune.anchors, dims = 1:20)
DefaultAssay(immune.combined) <- "integrated"

# Run the standard workflow for visualization and clustering
immune.combined <- ScaleData(immune.combined, verbose = FALSE)
immune.combined <- RunPCA(immune.combined, npcs = 30, verbose = FALSE)
# t-SNE and Clustering
immune.combined <- RunUMAP(immune.combined, reduction = "pca", dims = 1:20)
immune.combined <- FindNeighbors(immune.combined, reduction = "pca", dims = 1:20)
immune.combined <- FindClusters(immune.combined, resolution = 0.5)
# Visualization
p1 <- DimPlot(immune.combined, reduction = "umap", group.by = "stim")
p2 <- DimPlot(immune.combined, reduction = "umap", label = TRUE)
plot_grid(p1, p2)
DimPlot(immune.combined, reduction = "umap", split.by = "stim")

DefaultAssay(immune.combined) <- "RNA"
nk.markers <- FindConservedMarkers(immune.combined, ident.1 = 6, grouping.var = "stim", verbose = FALSE)
head(nk.markers)
FeaturePlot(immune.combined, features = c("CD3D", "SELL", "CREM", "CD8A", "GNLY", "CD79A", "FCGR3A", 
    "CCL2", "PPBP"), min.cutoff = "q9")

immune.combined <- RenameIdents(immune.combined, `0` = "CD14 Mono", `1` = "CD4 Naive T", `2` = "CD4 Memory T", 
    `3` = "CD16 Mono", `4` = "B", `5` = "CD8 T", `6` = "NK", `7` = "T activated", `8` = "DC", `9` = "B Activated", 
    `10` = "Mk", `11` = "pDC", `12` = "Eryth")

DimPlot(immune.combined, label = TRUE)
Idents(immune.combined) <- factor(Idents(immune.combined), levels = c("pDC", "Eryth", "Mk", "DC", 
    "CD14 Mono", "CD16 Mono", "B Activated", "B", "CD8 T", "NK", "T activated", "CD4 Naive T", "CD4 Memory T"))

markers.to.plot <- c("CD3D", "CREM", "HSPH1", "SELL", "GIMAP5", "CACYBP", "GNLY", "NKG7", "CCL5", 
    "CD8A", "MS4A1", "CD79A", "MIR155HG", "NME1", "FCGR3A", "VMO1", "CCL2", "S100A9", "HLA-DQA1", 
    "GPR183", "PPBP", "GNG11", "HBA2", "HBB", "TSPAN13", "IL3RA", "IGJ")
DotPlot(immune.combined, features = rev(markers.to.plot), cols = c("blue", "red"), dot.scale = 8, 
    split.by = "stim") + RotatedAxis()
theme_set(theme_cowplot())
t.cells <- subset(immune.combined, idents = "CD4 Naive T")
Idents(t.cells) <- "stim"
avg.t.cells <- log1p(AverageExpression(t.cells, verbose = FALSE)$RNA)
avg.t.cells$gene <- rownames(avg.t.cells)

cd14.mono <- subset(immune.combined, idents = "CD14 Mono")
Idents(cd14.mono) <- "stim"
avg.cd14.mono <- log1p(AverageExpression(cd14.mono, verbose = FALSE)$RNA)
avg.cd14.mono$gene <- rownames(avg.cd14.mono)

genes.to.label = c("ISG15", "LY6E", "IFI6", "ISG20", "MX1", "IFIT2", "IFIT1", "CXCL10", "CCL8")
p1 <- ggplot(avg.t.cells, aes(CTRL, STIM)) + geom_point() + ggtitle("CD4 Naive T Cells")
p1 <- LabelPoints(plot = p1, points = genes.to.label, repel = TRUE)
p2 <- ggplot(avg.cd14.mono, aes(CTRL, STIM)) + geom_point() + ggtitle("CD14 Monocytes")
p2 <- LabelPoints(plot = p2, points = genes.to.label, repel = TRUE)
plot_grid(p1, p2)
immune.combined$celltype.stim <- paste(Idents(immune.combined), immune.combined$stim, sep = "_")
immune.combined$celltype <- Idents(immune.combined)
Idents(immune.combined) <- "celltype.stim"
b.interferon.response <- FindMarkers(immune.combined, ident.1 = "B_STIM", ident.2 = "B_CTRL", verbose = FALSE)
head(b.interferon.response, n = 15)
FeaturePlot(immune.combined, features = c("CD3D", "GNLY", "IFI6"), split.by = "stim", max.cutoff = 3, 
    cols = c("grey", "red"))

```

# Applying mitch

The cell identities found in the tutorial are "pDC", "Eryth", "Mk", "DC", "CD14 Mono", "CD16 Mono", "B Activated", 
"B", "CD8 T", "NK", "T activated", "CD4 Naive T", "CD4 Memory T".
I will extract the differential expression data for each cell identity and then create a list object to feed into mitch.

```{r}
pDC <- FindMarkers(immune.combined, ident.1 = "pDC_STIM", ident.2 = "pDC_CTRL", verbose = FALSE)
Mk <- FindMarkers(immune.combined, ident.1 = "Mk_STIM", ident.2 = "Mk_CTRL", verbose = FALSE)
DC <- FindMarkers(immune.combined, ident.1 = "DC_STIM", ident.2 = "DC_CTRL", verbose = FALSE)
CD14mono <- FindMarkers(immune.combined, ident.1 = "CD14 Mono_STIM", ident.2 = "CD14 Mono_CTRL", verbose = FALSE)
CD16mono <- FindMarkers(immune.combined, ident.1 = "CD16 Mono_STIM", ident.2 = "CD16 Mono_CTRL", verbose = FALSE)
Bact <- FindMarkers(immune.combined, ident.1 = "B Activated_STIM", ident.2 = "B Activated_CTRL", verbose = FALSE)
B <- FindMarkers(immune.combined, ident.1 = "B_STIM", ident.2 = "B_CTRL", verbose = FALSE)
CD8T <- FindMarkers(immune.combined, ident.1 = "CD8 T_STIM", ident.2 = "CD8 T_CTRL", verbose = FALSE)
NK <- FindMarkers(immune.combined, ident.1 = "NK_STIM", ident.2 = "NK_CTRL", verbose = FALSE)
Tact <- FindMarkers(immune.combined, ident.1 = "T activated_STIM", ident.2 = "T activated_CTRL", verbose = FALSE)
CD4naiveT <- FindMarkers(immune.combined, ident.1 = "CD4 Naive T_STIM", ident.2 = "CD4 Naive T_CTRL", verbose = FALSE)
CD4memT <- FindMarkers(immune.combined, ident.1 = "CD4 Memory T_STIM", ident.2 = "CD4 Memory T_CTRL", verbose = FALSE)

x <- list("pDC"=pDC,"Mk"=Mk,"DC"=DC,"CD14mono"=CD14mono,"CD16mono"=CD16mono,"Bact"=Bact,"B"=B,"CD8T"=CD8T,
    "NK"=NK,"Tact"=Tact,"CD4naiveT"=CD4naiveT,"CD4memT"=CD4memT)

xx <- mitch_import(x = x,DEtype = "seurat")

download.file("https://reactome.org/download/current/ReactomePathways.gmt.zip", destfile="ReactomePathways.gmt.zip")
unzip("ReactomePathways.gmt.zip",overwrite = TRUE)
genesets <- gmt_import("ReactomePathways.gmt")
head(genesets,3)

capture.output(
    res <- mitch_calc(x = xx,genesets = genesets,priority = "effect",minsetsize=5)
    , file = "/dev/null", append = FALSE,
    type = c("output", "message"), split = FALSE)

head(res$enrichment_result,10)

unlink("seurat_mitch.html")
capture.output(
    mitch_report(res, "seurat_mitch.html")
    , file = "/dev/null", append = FALSE,
    type = c("output", "message"), split = FALSE)

```

```{r}
sessionInfo()
```
